src/Adapter/Product/Combination/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -164,20 +189,31 @@ private function formatEditableCombinationsForListing(
                 $combinationAttributesInformation,
                 (bool) $combination['default_on'],
                 $impactOnPrice,
-                (int) $this->stockAvailableRepository->getForCombination(new CombinationId($combinationId))->quantity
-// @todo:
-//      Missing combination image:
-//      Old page retrieves it through src/PrestaShopBundle/Controller/Admin/AttributeController::getFormImagesAction.
-//      we could simply get Product::getCombinationImageById and load new Image()->getbasePath,
-//      but not all combinations seems to have associated images with product
-//      (the old page still shows images for all of them - not sure if that is good behavior)
-//      also it is unclear how old page appends suffixes "small_default, home_default" (it seems controller only provides base path)
+                (int) $this->stockAvailableRepository->getForCombination(new CombinationId($combinationId))->quantity,
+                $this->getImagePath((int) $imageData['id_image'])
             );
         }
 
         return new CombinationListForEditing($totalCombinationsCount, $combinationsForEditing);
     }
 
+    /**
+     * @param int $imageId
+     *
+     * @return string
+     */
+    private function getImagePath(int $imageId): ?string
```

Comment:
I guess this part could also be in the ImageRepository? Or you think it's the handler responsibility?
If so maybe we need some Path factory service, I believe you made some for product images?

Reply:
building a path shouldn't be in a repository.
but some service maybe. Maybe I could reuse ImagePathFactory :thinking:

Reply:
Yeah it makes sense to build it in the handler, with the refacto of image factory it's smooth!

At first I thought the no picture should also be handled by the handler, but on second though it's more respectful of the domain to use `null` when no image is present and let the controller fill the blanks